src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersCardPage.tsx

```diff
@@ -1,84 +1,232 @@
             });
+        });
+        return {invidualCardsSectionData: individualCards, domainFeedsData: domainFeeds};
+    }, [filteredWorkspaceCardFeeds, newSelectedCards, styles.cardIcon, userCardList]);
+
+    const cardFeedsSectionData = useMemo(() => {
+        const repeatingBanks: string[] = [];
+        const banks: string[] = [];
+        const handleRepeatingBankNames = (bankName: string) => {
+            if (banks.includes(bankName)) {
+                repeatingBanks.push(bankName);
+            } else {
+                banks.push(bankName);
+            }
+        };
+
+        filteredWorkspaceCardFeeds.forEach(([cardFeedKey]) => {
+            const bankName = cardFeedKey.split('_').at(2);
+            if (!bankName) {
+                return;
+            }
+
+            handleRepeatingBankNames(bankName);
+        });
+        Object.values(domainFeedsData).forEach((domainFeed) => {
+            handleRepeatingBankNames(domainFeed.bank);
+        });
+
+        const cardFeedsData: CardFilterItem[] = [];
+
+        filteredWorkspaceCardFeeds.forEach(([, cardFeed]) => {
+            const representativeCard = Object.values(cardFeed ?? {}).find((cardFeedItem) => isCard(cardFeedItem));
+            if (!representativeCard) {
+                return;
+            }
+            const {domainName, bank} = representativeCard;
+            const isBankRepeating = repeatingBanks.includes(bank);
+            const cardFeedBankName = bank === CONST.EXPENSIFY_CARD.BANK ? translate('search.filters.card.expensify') : CardUtils.getCardFeedName(bank as CompanyCardFeed);
+            const policyID = domainName.match(CONST.REGEX.EXPENSIFY_POLICY_DOMAIN_NAME)?.[1] ?? '';
+            const correspondingPolicy = PolicyUtils.getPolicy(policyID?.toUpperCase());
+            const text = translate('search.filters.card.cardFeedName', {cardFeedBankName, cardFeedLabel: isBankRepeating ? correspondingPolicy?.name : undefined});
+            const correspondingCards = Object.keys(cardFeed ?? {}).filter((cardFeedKey) => cardFeedKey !== 'cardList');
+
+            cardFeedsData.push(buildCardFeedItem(text, policyID, correspondingCards, newSelectedCards, bank as CompanyCardFeed, styles.cardIcon));
+        });
+
+        Object.values(domainFeedsData).forEach((domainFeed) => {
+            const {domainName, bank, correspospondingCardIDs} = domainFeed;
+            const isBankRepeating = repeatingBanks.includes(bank);
+            const cardFeedBankName = bank === CONST.EXPENSIFY_CARD.BANK ? translate('search.filters.card.expensify') : CardUtils.getCardFeedName(bank as CompanyCardFeed);
+            const text = translate('search.filters.card.cardFeedName', {cardFeedBankName, cardFeedLabel: isBankRepeating ? domainName : undefined});
+
+            cardFeedsData.push(buildCardFeedItem(text, domainName, correspospondingCardIDs, newSelectedCards, bank as CompanyCardFeed, styles.cardIcon));
+        });
+        return cardFeedsData;
```

Comment:
We could rewrite `forEach` to a map

Reply:
We push to `cardFeedsData`, from two separate forEach loops `filteredWorkspaceCardFeeds.forEach` and `Object.values(domainFeedsData)`, so it's probably the cleanest way to do it.

Reply:
nvm :D